test/Server/SocketTest/SaveUploadTests.cs

```diff
@@ -0,0 +1,379 @@
+            {
+                result = result.Replace("\"guid\":\"\"", "\"guid\":\"" + guid + "\"");
+            }
+
+            return result;
+        }
+
+        void CheckHomeContent()
+        {
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.CodeBlock),
+                "Code block" + WebServerErrorMessages.NodeWasNotCreated);
+
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.Boolean),
+                "Boolean" + WebServerErrorMessages.NodeWasNotCreated);
+
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.If),
+                "If" + WebServerErrorMessages.NodeWasNotCreated);
+
+            Assert.IsTrue(Model.Nodes.First(node => node.GUID.ToString() == NodeGuids.CodeBlock).OutPorts.Count == 1,
+                WebServerErrorMessages.WrongCBNOutputPorts);
+        }
+
+        void CheckCustomNodeContent()
+        {
+            Assert.IsInstanceOf<CustomNodeWorkspaceModel>(Model.CurrentWorkspace);
+
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.Point),
+                "Point" + WebServerErrorMessages.NodeWasNotCreated);
+
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.Circle),
+                "Circle" + WebServerErrorMessages.NodeWasNotCreated);
+
+            Assert.IsTrue(Model.Nodes.Any(node => node.GUID.ToString() == NodeGuids.Surface),
+                "Surface" + WebServerErrorMessages.NodeWasNotCreated);
+        }
+
+        void CheckExecutingSaveMessage(string expectedFileName)
+        {
+            var messageHandler = new MessageHandler(Model);
+            string actualResult = null;
+
+            messageHandler.ResultReady += (s, args) =>
+            {
+                Assert.IsInstanceOf<SavedFileResponse>(args.Response, WebServerErrorMessages.WrongResponse);
+                var response = args.Response as SavedFileResponse;
+                actualResult = System.Text.Encoding.Default.GetString(response.FileContent.ToArray());
+            };
+
+            var commandPath = GetCommandPathByFileName("SaveFileMessage");
+            string saveCommand = File.ReadAllText(commandPath);
+
+            var message = messageHandler.DeserializeMessage(saveCommand);
+            messageHandler.Execute(message, "");
+
+            var resultPath = GetCommandPathByFileName(expectedFileName, "dyn");
+            var expectedResult = File.ReadAllText(resultPath);
+
+            Assert.NotNull(actualResult, WebServerErrorMessages.NoSaving);
+            Assert.AreEqual(expectedResult, actualResult, WebServerErrorMessages.WrongSavedFileContent);
+        }
+
+        void CheckExecutingUploadMessage(byte[] fileContent, int nodesNumber)
+        {
+            var messageHandler = new MessageHandler(Model);
+            bool proxyResponseTookPlace = false;
+
+            messageHandler.ResultReady += (s, args) =>
+            {
```

Comment:
Since `DynamoModel.StartInTestMode` is set to `true` for all `DynamoViewModelUnitTest` derived class, this will be called synchronously (i.e. no background thread that takes over execution and returns at a later time). So that is good. Just need to confirm that this callback is actually invoked (because without it invoked the tests will pass)?